bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.neohub.internal;
+
+import static org.openhab.binding.neohub.internal.NeoHubBindingConstants.*;
+
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link NeoHubHandler} is the OpenHab Handler for NeoHub devices
+ *
+ * @author Andrew Fiddian-Green - Initial contribution (v2.x binding code)
+ * @author Sebastian Prehn - Initial contribution (v1.x hub communication)
+ * 
+ */
+public class NeoHubHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(NeoHubHandler.class);
+
+    private NeoHubConfiguration config;
+    private NeoHubSocket socket;
+
+    private ScheduledFuture<?> lazyPollingScheduler;
+    private ScheduledFuture<?> fastPollingScheduler;
+
+    private final AtomicInteger fastPollingCallsToGo = new AtomicInteger();
+
+    public NeoHubHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // future: currently there is nothing to do for a NeoHub
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(NeoHubConfiguration.class);
+
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
```

Comment:
Should be CONFIGURATION_ERROR - list the missing params in the text message.

Reply:
Ok, done.